### DISCUSSION TITLE
live_grep jumps to beginning of the file on first search after nvim open

### DISCUSSION THREAD

**Discussion by kapral18** [LazyVim/LazyVim#1606] [Q&A] [answered]
my config: https://github.com/kapral18/config.nvim/blob/7327cd3005c3572748bd3153001a09b02e696f03/lua/config/lazy.lua

I use default lazyvim bindings for <leader>sg to live_grep the text, but for whatever reason after initial nvim open when I hit the grep result it jumps to the beginning of the page instead of the actual match, after that it works ok. But every time I close and open nvim it happens again.

Anyone else experienced the same?

**Comment by dpetka2001** [comment]
Not having this problem. Whenever you have problems like this, it's good to test things on a fresh Lazyvim starter template to make sure if it's a Lazyvim problem or a problem with your configuration.

**Reply by kapral18** [reply]
thanks for checking

**Comment by folke** [comment]
@kapral18 that was fixed yesterday. Are you using the latest version of LazyVim/lazy.nvim?

**Reply by kapral18** [reply]
I just checked and can confirm, thank you!!!

**Reply by kapral18** [reply]
hey @folke it somehow broke again. Now instead of jumping to THE beginning of the file it jumps somewhere in the vicinity of the beginning of the file.

**Reply by kapral18** [reply]
tbh it looks like it's sometimes working and sometimes not I cannot pinpoint the breaking point

**Reply by folke** [reply]
I don't have this. Are you sure you are using the latest version of LazyVim and lazy.nvim?

Wouldbe great if you can post the steps that are reproducible.

**Reply by kapral18** [reply]
It's already in my muscle memory to `<leader>l` and `U` :) so 100%. 

For example you can clone my latest config `https://github.com/kapral18/config.nvim/blob/ffdb572892a01f74b7328cc8279008136258baf0/lazy-lock.json`, and then clone this project `https://github.com/kapral18/karenjs.com` and spin up nvim and `<leader>sg` for `perfectionkills`. It should jump you to line 1741 in index.md but it jumps me to 1718

And it's not confined to specific extension, it happens on .ts or .py files for me, and the offset is random. Sometimes it works from the get go but other times it randomly jumps to a different location.

**Reply by folke** [reply]
Just tried with your config and that repo and it goes to the correct spot in the doc

**Reply by kapral18** [reply]
@folke can you try mulitple times the same seach but the key is to kill nvim instance and start a new each time before search.

**Reply by folke** [reply]
Just did it over 10 times. Always opens at the correct location

**Reply by kapral18** [reply]
@folke https://asciinema.org/a/HcKTtYtNfSsvEZ4MZfRGrhDnJ

**Reply by kapral18** [reply]
ok lemme try with docker

**Reply by kapral18** [reply]
yess I tried in docker and it's even more reliably mistagreted there, can you try in docker?

**Reply by kapral18** [reply]
basically I did enter the folder with karenjs.com/ project and then 

```sh
docker run -w /root -v ./:/root/karenjs.com/ -it --rm alpine:edge sh -uelic '
    apk add git lazygit neovim ripgrep alpine-sdk --update
    git clone https://github.com/kapral18/config.nvim.git ~/.config/nvim && sh
  '
```

then inside container I did `nvim` once to install everything, after it was done I exited and opened again to just be sure and tried to search diff things, nothing matched :(

**Reply by folke** [reply]
ok, can reproduce now. The trigger is opening neovim with `nvim .`

**Reply by kapral18** [reply]
@folke I see. Is it ok to use nvim like that or?

**Reply by folke** [reply]
yep, that should work. Looking into it. It seems that files opened through Telescope trigger `BufReadPost` twice.
IT's an easy fix for last_loc autocmd, but I first want to be sure it's not an issue with lazy.nvim.

**Reply by folke** [reply]
It's caused by the neo-tree spec. Have to go here now, but will fix it later this evening

**Reply by kapral18** [reply]
@folke tbh I noticed a lot of weird glitches happen when neo-tree is open. Thanks btw

**Reply by kapral18** [reply]
@folke since you already opened my project in docker do you also see this weird warning in markdown files?

![CleanShot 2023-10-09 at 19 20 57@2x](https://github.com/LazyVim/LazyVim/assets/1625373/b9ccc9ac-8342-47aa-bc48-73d96737e46e)

Also for some reason when any notifications trigger, my neovim goes into insane lags mode, regardless of error or warning.

**Reply by folke** [reply]
Yep, saw that too. Copilot also freaks out on that project for some reason. Will see what's wrong there

**Reply by folke** [reply]
I know what the issue is. It only happens when doing `nvim .` and when you have your focus on `neo-tree`.

- Then opening a file with telescope, opens the buffer in the neo-tree window, triggering `BufReadPost`
- Neo-tree then reloads itself in the window and loads the buffer in the main window instead, again triggering a `BufReadPost` event.

So all good on the lazy and LazyVim side except for the last_loc autocmd itself. Wil push a fix in a bit

**Reply by folke** [reply]
It's not something I can fix. It's pretty much a bug between telescope and neo-tree.
It may possibly be fixed in neo-tree though, by also setting the correct cursor at:
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/6f8c49956c89e9fefae6acdfe1d57c6293b0a03d/lua/neo-tree/setup/init.lua#L256

Best to open an issue at the neo-tree repo.

**Reply by folke** [reply]
https://github.com/nvim-neo-tree/neo-tree.nvim/issues/958

**Reply by folke** [reply, marked as answer]
So I was able to fix this after all with the `get_selection_window` telescope option.

The telescope config now has a better `get_selection_window` that returns the first file window if it exists.
When the edgy extra is enabled, then edgy is used instead, since edgy knows exactly what are main windows and which ones aren't.

Will push a new release later today.

**Reply by folke** [reply]
That `null-ls` error is simply because you didnt install the local node modules and because of your prettier setting:

```lua
        nls.builtins.formatting.prettier.with({
          prefer_local = "node_modules/.bin",
        })

```

**Reply by kapral18** [reply]
@folke 
![CleanShot 2023-10-09 at 21 01 14@2x](https://github.com/LazyVim/LazyVim/assets/1625373/a3a1567b-099f-4f63-a09e-5b6db2d5980c)

in all examples node_modules and prettier are installed

**Reply by folke** [reply]
still that part of the code though. removing the prefer_local no longer shows the error

**Reply by kapral18** [reply]
@folke I see, thanks for the hint
